### PR TITLE
pathogen-repo-ci: Multi-runtime

### PR DESCRIPTION
**Pull request by tsibley**
See commit messages.

### Testing

- [x] Checks pass

### CODE REVIEW

**Comment by tsibley** [comment]
This might reveal bugs in our Conda runtime for ncov and seasonal-flu: https://github.com/nextstrain/conda-base/pull/27#issuecomment-1544457814

File: .github/workflows/pathogen-repo-ci.yaml [hunk 1]
```diff
@@ -44,10 +44,29 @@ on:
         default: ""
         required: false
 
+      runtimes:
+        description: >-
+          List of Nextstrain runtimes under which to run the build, as a string
+          containing YAML.  This is easily produced, for example, by pretending
+          you're writing normal nested YAML within a literal multi-line block
+          scalar (introduced by "|"):
```
Comment (victorlin) [inline .github/workflows/pathogen-repo-ci.yaml:52]: Neat workaround for input type limitations.

**Review by victorlin — APPROVED** [review]

File: .github/workflows/pathogen-repo-ci.yaml [hunk 2]
```diff
@@ -56,7 +88,25 @@ permissions:
...
+          runtimes="$(yq --output-format=json --indent=0 . <<<"$runtimes")"
+          echo runtimes="$runtimes" | tee -a "$GITHUB_OUTPUT"
+    outputs:
+      runtimes: ${{ steps.inputs.outputs.runtimes }}
+
   build:
+    needs: configuration
+    strategy:
+      matrix:
+        runtime: ${{ fromJSON(needs.configuration.outputs.runtimes) }}
```
Comment (joverlee521) [inline .github/workflows/pathogen-repo-ci.yaml:108]: Thoughts on adding `fail-fast: false` so that runtime failures do not interfere with each other?

Comment (victorlin) [reply .github/workflows/pathogen-repo-ci.yaml:108]: Sounds like a good idea!

Comment (tsibley) [reply .github/workflows/pathogen-repo-ci.yaml:108]: Yep, that makes sense. Would you do it?

**Pull request merged**